yearn/treasury/treasury.py

```diff
@@ -0,0 +1,319 @@
+import logging
+import threading
+import time
+
+from brownie import Contract, chain, web3
+from brownie.network.event import EventLookupError
+from eth_abi import encode_single
+from joblib import Parallel, delayed
+from yearn.events import decode_logs
+from yearn.multicall2 import fetch_multicall
+from yearn.outputs import victoria
+from yearn.partners.partners import partners
+from yearn.partners.snapshot import WildcardWrapper, Wrapper
+from yearn.prices.constants import weth
+from yearn.prices.magic import PriceError, get_price, logger as logger_price_magic
+from ypricemagic.utils.utils import Contract_with_erc20_fallback
+
+from ..constants import TREASURY_WALLETS
+
+logger = logging.getLogger(__name__)
+logger_price_magic.setLevel(logging.CRITICAL)
+
+
+def _get_price(token, block=None):
+    SKIP_PRICE = [  # shitcoins
+        "0xa9517B2E61a57350D6555665292dBC632C76adFe",
+        "0xb07de4b2989E180F8907B8C7e617637C26cE2776",
+        "0x1368452Bfb5Cd127971C8DE22C58fBE89D35A6BF",
+        "0x5cB5e2d7Ab9Fd32021dF8F1D3E5269bD437Ec3Bf",
+        "0x11068577AE36897fFaB0024F010247B9129459E6",
+        "0x9694EED198C1b7aB81ADdaf36255Ea58acf13Fab",
+        "0x830Cbe766EE470B67F77ea62a56246863F75f376",
+        "0x8F49cB69ee13974D6396FC26B0c0D78044FCb3A7",
+        "0x53d345839E7dF5a6c8Cf590C5c703AE255E44816",
+        "0xcdBb37f84bf94492b44e26d1F990285401e5423e",
+        "0xE256CF1C7caEff4383DabafEe6Dd53910F97213D",
+        "0x528Ff33Bf5bf96B5392c10bc4748d9E9Fb5386B2",
+    ]
+    if token in SKIP_PRICE:
+        return 0
+    
+    try:
+        price = get_price(token, block)
+    except AttributeError:
+        logger.warn(
+            f"AttributeError while getting price for {Contract(token).symbol()} {token}"
+        )
+        raise
+    except PriceError:
+        logger.warn(
+            f"PriceError while getting price for {Contract(token).symbol()} {token}"
+        )
+        price = 0
+    except ValueError:
+        logger.warn(
+            f"ValueError while getting price for {Contract(token).symbol()} {token}"
+        )
+        price = 0
+    return price
+
+
+def get_token_from_event(event):
+    try:
+        return event['Transfer'][0].address
+    except EventLookupError:
+        logger.critical(
+            f'One of your cached contracts has an incorrect definition: {event.address}. Please fix this manually'
+        )
+        raise Exception(
+            f'One of your cached contracts has an incorrect definition: {event.address}. Please fix this manually'
+        )
+
+
+class Treasury:
+    '''
+    Used to export Yearn financial reports
+    '''
+
+    def __init__(self, watch_events_forever=False):
+        self.addresses = list(TREASURY_WALLETS)
+        self._transfers = []
+        self._topics_in = [
+            '0xddf252ad1be2c89b69c2b068fc378daa952ba7f163c4a11628f55a4df523b3ef',
+            None,
+            ['0x000000000000000000000000' + address[2:] for address in self.addresses],
```

Comment:
alternatively here you can use `eth_utils.encode_hex(eth_abi.encode_single('address', address)`